photo/queries.py

```diff
@@ -37,49 +40,49 @@ def pictures(self, picture: int = None) -> List[PictureType]:
 
     @strawberry.field
     def picture_comments(
-        self, id: int = None, user_email: str = None, picture_id: int = None
+        self, id: int = None, user: uuid.UUID = None, picture_id: int = None
     ) -> List[PictureCommentType]:
         if id:
             return PictureComment.objects.filter(id=id)
-        if user_email and picture_id:
-            return PictureComment.objects.filter(user=user_email, picture=picture_id)
-        elif user_email:
-            return PictureComment.objects.filter(user=user_email)
+        if user and picture_id:
+            return PictureComment.objects.filter(user=user, picture=picture_id)
+        elif user:
+            return PictureComment.objects.filter(user=user)
         elif picture_id:
             return PictureComment.objects.filter(picture=picture_id)
         return PictureComment.objects.all()
 
     @strawberry.field
     def collections(
-        self, user_email: str = None, name: str = None, id: int = None
+        self, user: uuid.UUID = None, name: str = None, id: int = None
     ) -> List[CollectionType]:
         if id:
             return Collection.objects.filter(id=id)
-        if user_email and name:
-            return Collection.objects.filter(user__email=user_email, name=name)
-        elif user_email:
-            return Collection.objects.filter(user__email=user_email)
+        if user and name:
+            return Collection.objects.filter(user=user, name=name)
+        elif user:
+            return Collection.objects.filter(user=user)
         elif name:
             return Collection.objects.filter(name=name)
         return Collection.objects.all()
 
     @strawberry.field
-    def contests(self, user_email: str = None, id: int = None) -> List[ContestType]:
+    def contests(self, user: uuid.UUID = None, id: int = None) -> List[ContestType]:
         if id:
             return Contest.objects.filter(id=id)
-        if user_email:
-            return Contest.objects.filter(created_by=user_email)
+        if user:
+            return Contest.objects.filter(created_by=user)
```

Comment:
this should only do `return Contest.objects.all()`

Reply:
Implemented